libs/environment/src/hooks/use-environment.ts

```diff
@@ -600,35 +618,31 @@ export const useEnvironment = create<EnvStore>()((set, get) => ({
     }
 
     const state = get();
-    const storedUrl = LocalStorage.getItem(STORAGE_KEY);
+
+    let storedUrl = LocalStorage.getItem(STORAGE_KEY);
+    if (!isValidUrl(storedUrl)) {
+      // remove invalid data from local storage
+      LocalStorage.removeItem(STORAGE_KEY);
+      storedUrl = null;
+    }
 
     let nodes: string[] | undefined;
     try {
-      nodes = await fetchConfig(state.VEGA_CONFIG_URL);
-      const enrichedNodes = uniq(
-        [...nodes, state.VEGA_URL, storedUrl].filter(Boolean) as string[]
+      nodes = uniq(
+        compact([
+          // url from local storage
+          storedUrl,
+          // url from state (if set via env var)
+          state.VEGA_URL,
+          // urls from network configuration
+          ...(await fetchConfig(state.VEGA_CONFIG_URL)),
+        ])
       );
-      set({ nodes: enrichedNodes });
+      set({ nodes });
     } catch (err) {
       console.warn(`Could not fetch node config from ${state.VEGA_CONFIG_URL}`);
     }
 
-    // Node url found in localStorage, if its valid attempt to connect
-    if (storedUrl) {
-      if (isValidUrl(storedUrl)) {
-        set({ VEGA_URL: storedUrl, status: 'success' });
-        return;
-      } else {
-        LocalStorage.removeItem(STORAGE_KEY);
-      }
-    }
-
-    // VEGA_URL env var is set and is a valid url no need to proceed
-    if (state.VEGA_URL) {
```

Comment:
@asiaznik @edd  gov and explorer have set VEGA_URLs at build. The idea here was that if VEGA_URL is set at build time we just use that node immediately, without checking other nodes.

So this change I believe will make gov and explorer start searching for nodes at startup. Which is probably fine, but just wanted to highlight this

Reply:
@mattrussell36 this sounds fine, thanks for highlighting it

Reply:
It's a delibarate feature :feelsgood: